### PR TITLE
Add support for named imports in ESM

### PR DESCRIPTION
**Pull request by shadowspawn**
# Pull Request

## Problem

Importing commander as CommonJs into an ECMAScript module only allows using the default import, and can not use named imports.

```js
// import { program } from 'commander'; // ERROR
import commander from 'commander';
const program = commander.program;
```

Related issue: #1284

## Solution

Add esm wrapper and use as a deep import:

```
import { program } from 'commander/esm.mjs';
```

In the future we will add support so the deep import is not needed by using conditional exports and/or subpaths. Being conservative for now as the functionality is still marked as experimental in node 12. (There have been breaking changes in the support in minor versions of node 12 and 13.) 

- https://nodejs.org/docs/latest-v12.x/api/packages.html#packages_conditional_exports
- https://nodejs.org/docs/latest-v12.x/api/packages.html#packages_subpath_exports

The `--experimental-modules` option used in the `test-esm` run-script will get dropped by node eventually, but currently allows that test to run locally with node 10 through node 15.

## ChangeLog

- added support for named imports from ECMAScript modules

### CODE REVIEW

**Review by abetomo — APPROVED** [review]
LGTM

**Comment by shadowspawn** [comment]
So fast! 🚀 
Thanks @abetomo

File: package.json [hunk 1]
```diff
@@ -19,18 +19,21 @@
...
     "test-typings": "tsc -p tsconfig.json",
     "typescript-checkJS": "tsc --allowJS --checkJS index.js --noEmit",
-    "test-all": "npm run test && npm run lint && npm run typescript-lint && npm run typescript-checkJS"
+    "test-all": "npm run test && npm run lint && npm run typescript-lint && npm run typescript-checkJS && npm run test-esm"
   },
-  "main": "index",
+  "main": "./index.js",
   "files": [
     "index.js",
+    "wrapper.mjs",
```
Comment (cravler) [inline package.json:33]: looks like a typo, must be `esm.mjs`

Comment (shadowspawn) [reply package.json:33]: Oh, good catch, thanks! I changed the name half way through.

Comment (shadowspawn) [reply package.json:33]: (For interest, I thought `wrapper` was a suitable name for a wrapper used with conditional exports and not seen by normal client use. But when I changed to deep import and the name became visible, I wanted something more meaningful.)

Comment (shadowspawn) [reply package.json:33]: Fixed in #1443

**Pull request merged**